specs/Deployment/MSIXPackages.md

```diff
@@ -0,0 +1,189 @@
+
+The [Dynamic Dependency API](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+resolves a package dependency to a specific framework package and make it available for a process,
+but Windows doesn't know the package is in use by this process. Lacking this "package in use"
+knowledge, Windows could service the package (remove, remediate, repair, ...) in negatively
+impactful ways. To prevent this (without servicing Windows) we need...
+
+1. ...a running process running with package identity...
+2. ...with an `appxmanifest.xml` declaring a `<PackageDependnecy>` on the framework package...
+3. ...running for as long as the framework package is used by the app.
+
+This is accomplished for the Microsoft.WindowsAppRuntime framework package via a Packaged COM
+out-of-process (OOP) server in a DDLM package, selected and managed via the Bootstrap API. See the
+[Dynamic Dependency spec](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+for more details.
+
+## Naming
+
+Windows App SDK's MSIX packages use the following naming rules for package identity:
+
+* Name = Microsoft.WindowsAppRuntime[.SubName].<Major>.<Minor>[-VersionTag]
+* Publisher = "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"
+
+where
+
+* SubName -- optional package sub-name. Use to distinguish amongst the different packages
+* Major -- major version of the release, e.g. "1" for Windows App SDK 1.0
+* Minor -- minor version of the release, e.g. "0" for Windows App SDK 1.0
+* VersionTag -- optional version tag to distinguish amongst channels and revs of a channel, e.g. "-preview2" for Windows App SDK 1.0 Preview 2
+
+### Naming - SubName
+
+The following SubName values are used:
+
+| SubName | Package | Example |
+|-|-|-|
+| | Framework | Microsoft.WindowsAppRuntime.1.0-experimental1 |
+| Main | Main | Microsoft.WindowsAppRuntime.Main.1.0-experimental1 |
+| Singleton | Singleton | Microsoft.WindowsAppRuntime.Singleton.1.0-experimental1 |
+| DDLM | Dynamic Dependency Lifetime Manager (DDLM) | Microsoft.WindowsAppRuntime.DDLM.1.0-experimental1 |
+
+### Naming - Singleton
+
+The Singleton package follows a different naming scheme
+
+* Name = Microsoft.WindowsAppRuntime.Singleton
+
+The package family is the same across all versions of Windows App SDK. As a main package, only one
+Singleton package can be registered at a time for a user. This will be the highest version of all
+installed, e.g. if Windows App SDK 1.0, 1.1 and 2.0 (Stable) are installed on an x86 system, the user will
+have registered...
+
+* Microsoft.WindowsAppRuntime.1.0
+* Microsoft.WindowsAppRuntime.1.1
+* Microsoft.WindowsAppRuntime.2.0
+* Microsoft.WindowsAppRuntime.Main.1.0
+* Microsoft.WindowsAppRuntime.Main.1.1
+* Microsoft.WindowsAppRuntime.Main.2.0
+* Microsoft.WindowsAppRuntime.Singleton (version 2.0)
+* Microsoft.WindowsAppRuntime.DDLM.1.0-x8
```

Comment:
what's the reason for "x8" instead of "x86"? not intuitive with one char saving.

Reply:
AppExtension name is limited to <= 39 characters on builds less than 18307 and the WAS min target is RS5 17763. When the min moves forward, presumably all this can be refactored away.

Reply:
Length. A package's Name is limited to <= 50 characters, so DDLM is cramped for space. I see the detailed naming rules aren't posted; let me dig those up.

And actually the DDLM here is wrong. DDLM names are actually `Microsoft.ProjectReunion.DDLM-<version>-<shortarchitecture>[-shortversiontag]` where `<version>` is the full MSIX version (not the project's major.minor release numbering). For instance, even though we released the  `Microsoft.ProjectReunion.0.8-preview1` framework package the associated x64 DDLM package had a Name of `Microsoft.ProjectReunion.DDLM.8000.146.628.0-x6-p`. 

(Soon to be renamed for WindowsAppRuntime, though DDLM will need be Microsoft.WinAppRuntime.DDLM... to fit within name limits)